File: Data Structures & Algorithms/maximum-number-of-balloons/submission-1.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        from collections import Counter

        freq = Counter(text)

        # Each balloon uses: b(1), a(1), l(2), o(2), n(1)
        return min(
            freq.get('b', 0),
            freq.get('a', 0),
            freq.get('l', 0) // 2,
            freq.get('o', 0) // 2,
            freq.get('n', 0)
        ) 

File: Data Structures & Algorithms/perfect-squares/submission-1.cpp
class Solution {
public:
    int numSquares(int n) {
        vector<int> dp(n+1,INT_MAX);
        dp[0] = 0;
        dp[1] = 1;

        for(int i = 1; i <= n ; i++)
        {
            int target = i;
            for(int j = 1; j * j <= target ; j++ )
            {
                dp[target] = min(dp[target], 1 + dp[target - j*j]);
            }
        }

        return dp[n];
    }
};